plugin/pkg/auth/authorizer/node/node_authorizer.go

```diff
@@ -335,6 +346,79 @@ func (r *NodeAuthorizer) authorizeResourceSlice(nodeName string, attrs authorize
 	return r.authorize(nodeName, sliceVertexType, attrs)
 }
 
+// authorizeNode authorizes node requests to Node API objects
+func (r *NodeAuthorizer) authorizeNode(nodeName string, attrs authorizer.Attributes) (authorizer.Decision, string, error) {
+	switch attrs.GetSubresource() {
+	case "":
+		switch attrs.GetVerb() {
+		case "create", "update", "patch":
+			// Use the NodeRestriction admission plugin to limit a node to creating/updating its own API object.
+			return authorizer.DecisionAllow, "", nil
+		case "get", "list", "watch":
+			return r.authorize(nodeName, nodeVertexType, attrs)
+		}
+	case "status":
+		switch attrs.GetVerb() {
+		case "update", "patch":
+			// Use the NodeRestriction admission plugin to limit a node to updating its own Node status.
+			return authorizer.DecisionAllow, "", nil
+		}
+	}
+
+	klog.V(2).Infof("NODE DENY: '%s' %#v", nodeName, attrs)
+	return authorizer.DecisionNoOpinion, "", nil
+}
+
+// authorizePod authorizes node requests to Pod API objects
+func (r *NodeAuthorizer) authorizePod(nodeName string, attrs authorizer.Attributes) (authorizer.Decision, string, error) {
+	switch attrs.GetSubresource() {
+	case "":
+		switch attrs.GetVerb() {
+		case "get":
+			return r.authorizeGet(nodeName, podVertexType, attrs)
+
+		case "list", "watch":
+			// allow a scoped fieldSelector
+			reqs, _ := attrs.GetFieldSelector()
+			for _, req := range reqs {
+				if req.Field == "spec.nodeName" && req.Operator == selection.Equals && req.Value == nodeName {
+					return authorizer.DecisionAllow, "", nil
+				}
```

Comment:
we always use equals, not in?

Reply:
this is the actual effective parsed field selector, which only supports Equals and NotEquals:

https://github.com/kubernetes/kubernetes/blob/f2428d66cc2256fcbe2a486b1403603530f46c6f/staging/src/k8s.io/apimachinery/pkg/fields/selector.go#L447-L452

https://github.com/kubernetes/kubernetes/blob/f2428d66cc2256fcbe2a486b1403603530f46c6f/staging/src/k8s.io/apimachinery/pkg/fields/selector.go#L110-L113

https://github.com/kubernetes/kubernetes/blob/f2428d66cc2256fcbe2a486b1403603530f46c6f/staging/src/k8s.io/apimachinery/pkg/fields/selector.go#L158-L161

In/NotIn/Exists/DoesNotExist are not a thing yet for the internal parsed field selector.